app/model/profile.ts

```diff
@@ -195,27 +209,60 @@ export class ProfileRecord extends Realm.Object implements ProfileRecordRaw {
 
     try {
       const { profileName = UNTITLED_PROFILE_NAME } = profileMetadata?.data ?? {};
+      
+      // Check if profile with same name already exists (case-insensitive)
+      const existingProfiles = await ProfileRecord.getAllProfileRecords();
+      const existingProfile = existingProfiles.find(profile => isProfileNameDuplicate(profile.profileName, profileName));
+      
+      if (existingProfile) {
+        // Skip profile creation but still process credentials for existing profile
+        const profileRecordId = existingProfile._id;
+        profileImportReport.userIdImported = false; // Profile already exists
+        profileImportReport.profileDuplicate = true;
+        
+        const existingCredentials = await CredentialRecord.getAllCredentialRecords();
+        const profileCredentialIds = existingCredentials
+          .filter(({ profileRecordId: credProfileId }) => credProfileId.equals(profileRecordId))
+          .map(({ credential }) => credential.id);
+
+        await Promise.all(
+          credentials.map(async (credential) => {
+            let achievement = credential.credentialSubject.hasCredential ?? credential.credentialSubject.achievement;
+            if (Array.isArray(achievement)) {
+              achievement = achievement[0];
+            }
+            const credentialName = achievement?.name ?? 'Unknown Credential';
```

Comment:
This logic should be extracted into `getCredentialName()` function, and add a unit test or two for it.

Also change https://github.com/openwallet-foundation-labs/learner-credential-wallet/blob/main/app/lib/credentialDisplay/openBadgeCredential.tsx#L60 to use this new function. (In general, anywhere we are determining the credential name, let's use the function, since it's a nontrivial operation).